tests/functional/generate_pages.py

```diff
@@ -409,12 +409,47 @@ def process_calculated_summary(answers, page_spec):
         page_spec.write(CALCULATED_SUMMARY_LABEL_GETTER.substitute(answer_context))
 
 
-def process_final_summary(schema_data, page_spec, collapsible, section_summary=False):
-    for section in schema_data["sections"]:
-        write_summary_spec(collapsible, page_spec, section, section_summary)
+def process_final_summary(
+    schema_data, require_path, dir_out, spec_file, collapsible, section_summary=False
+):
+    logger.debug("Processing questionnaire flow")
+
+    page_filename = f"summary.page.js"
+    page_path = os.path.join(dir_out, page_filename)
+
+    logger.info("creating %s...", page_path)
 
-    if collapsible:
-        page_spec.write(COLLAPSIBLE_SUMMARY_GETTER)
+    with open(page_path, "w") as page_spec:
+        page_name = "FinalSummary"
+        base_page = "QuestionPage"
+        base_page_file = "question.page"
```

Comment:
Inheriting from QuestionPage seems wrong - it has a lot of methods that aren't relevant.

Reply:
Base pages have been slightly refactored and this now inherits from `SubmitBasePage`.